Simplified/AxisService/NYPLAxisContentDownloader.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  NYPLAxisContentDownloader.swift
+//  Simplified
+//
+//  Created by Raman Singh on 2021-04-22.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+
+protocol NYPLAxisContentDownloading {
+  func downloadItem(from url: URL,
+                    _ completion: @escaping (Result<Data, Error>) -> Void)
+}
+
+
+/// A content downloader which stops downloading content upon receiving a failure.
+class NYPLAxisContentDownloader: NYPLAxisContentDownloading {
+  
+  private let tasks: ThreadSafeDictionaryContainer<URL, URLSessionDataTask>
```

Comment:
do these properties need to stay consistent with each other?  the container is nice but will only ensure thread safety of each individual instance, since the queue it's using to synchronize access is different for each one.

Reply:
These properties are supposed to stay consistent. Now that you mentioned it, I think we should have a single source of truth instead of relying on multiple objects . A single class that stores all these values (handlres, tasks, errored) should be more appropriate. Let me know if that's something you'd prefer and we can go with that.

Reply:
there are many ways to achieve the same thing, your approach seem fine. In a way you already have a class that store all the values (the Downloader itself) you could just protect access with a NSRecursiveLock... honestly the simplest is perhaps to parametrize your nice container classes on the same queue. This just needs the ThreadContainers to receive a queue from the caller in the initializer, and you're done I think...

Reply:
Done